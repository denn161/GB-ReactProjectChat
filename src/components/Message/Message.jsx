

import React from 'react';
import PropTypes from 'prop-types';
import styles from './Message.module.css';




const Message = ({ text,flag}) => {   
    
    return (    
        
      <div >  
      <h1 className={styles.text} style={{color:flag?"orange":"darkgreen"}}>{text}</h1>   
      </div>
  );
}


Message.propTypes = {
  text: PropTypes.string
}
export default Message;