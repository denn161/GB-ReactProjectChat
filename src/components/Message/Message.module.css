
.text{
   font-weight: 700;
   font-size: 26px;
   

}
.message{
   border-radius: 20px;
}