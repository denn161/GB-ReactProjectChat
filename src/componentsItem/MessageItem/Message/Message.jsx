import PropTypes from 'prop-types';
import close from './close/close.png'
import styles from './Message.module.css';


const Message = ({message,removeMessage}) => {
  return (
    <li className={styles.item} style={{
        margin: 10,
        border:"2px solid green",
        marginLeft:"auto",
        width: "fit-content",
        padding: 20,borderRadius:10                        }}>
      <span>{message.name}</span>
        <p>{message.message}</p>
       <img onClick={()=>removeMessage(message.id)} src={close} alt="close" className={styles.btn} />
        </li>
  );
}

Message.propTypes = {
    message: PropTypes.object,
    removeMessage:PropTypes.func
}
export default Message;