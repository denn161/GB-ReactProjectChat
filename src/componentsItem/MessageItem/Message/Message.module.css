

.item{
  position:relative;
}

.btn{
  width: 20px;
  height: 20px;
  position: absolute;
  top: 6px;
  right:6px;
  border: none;
 border-radius: 10px;
}