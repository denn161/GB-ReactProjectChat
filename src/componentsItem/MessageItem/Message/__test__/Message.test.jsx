import { render, screen } from '@testing-library/react';
import Message from '..';


describe('Message tests', () => {
    
    it('renders name and message', () => {
     
        const user = {name: 'User', message: 'messageText'}
        
     render(<Message message={user} />)

     const msgText = screen.getByText('messageText')
     const authorText = screen.getByText('User')
     
      expect(msgText).toBeInTheDocument()
      expect(authorText).toBeInTheDocument()

})

it('matches snapshot', () => {
   const user = {name: 'User', message: 'messageText'}
   const message = render(<Message message={user} />)
    
   expect(message).toMatchSnapshot()     

})

})