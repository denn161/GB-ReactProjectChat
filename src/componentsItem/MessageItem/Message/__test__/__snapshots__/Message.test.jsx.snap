// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message tests matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="item"
        style="margin: 10px 10px 10px auto; border: 2px solid green; padding: 20px; border-radius: 10px;"
      >
        <span>
          User
        </span>
        <p>
          messageText
        </p>
        <img
          alt="close"
          class="btn"
          src="close.png"
        />
      </li>
    </div>
  </body>,
  "container": <div>
    <li
      class="item"
      style="margin: 10px 10px 10px auto; border: 2px solid green; padding: 20px; border-radius: 10px;"
    >
      <span>
        User
      </span>
      <p>
        messageText
      </p>
      <img
        alt="close"
        class="btn"
        src="close.png"
      />
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
